### ISSUE TITLE
Maintained?

### ISSUE THREAD

**Issue by MggMuggins** [7h0ma5/redox-ssh#3]
We've been talking in the redox developer chat (@7h0ma5 are you a member?) and ssh has come up a couple times recently. Is this project still maintained? I'd been thinking of doing some work with trying to get pure-rust ssh running under redox myself, are you interested in a hand? I'm gonna do a little more in-depth assessment of this repo and maybe start hacking on it.

**Comment by ERnsTL** [comment]
Redox can access the internet fine now since 0.3.5 so an SSH server and client are the hacker's obvious next requirements.

I would also hack on this if I knew it's maintained - at least timely accepting patches and PRs would be enough.

Otherwise a fork to a community project like github.com/redox-ssh/redox-ssh would also be possible.

**Comment by ERnsTL** [comment]
Greetings, I have received response from the author and relay it here:

Future: He is happy that there is interest in the project, but does neither now nor will have in the future the time to maintain it. He would thus be happy if someone would take it over.

Licensing: He could transfer copyright to "the Redox OS developers" or an organization or anyone who would maintain it. License change to MIT would be possible. Several Rust projects I have seen use dual Apache2 + MIT (template: [limn GUI library](https://github.com/christolliday/limn#license)). I am personally also fine with it staying Lesser GPL.

So far the message from the author; how shall we proceed?

I personally would favor a proper Github organization where multiple authors can contribute and project continuity is ensured.

**Comment by MggMuggins** [comment]
I think it would make most sense to transfer copyright to Redox-OS and transfer this repo to the redox github organization. If the project gets big enough we could shift it to it's own Org.

I'm a big fan of MIT, I'm not a huge fan of the GPL. If we transfer to Redox-OS I think @jackpot51 would probably be the one to make the licensing call. He would also _probably_ grant us write access to the repo, in addition to all members of Redox's dev team.

**Comment by jackpot51** [comment]
@ERnsTL this would be great! We would certainly maintain it!

**Comment by ERnsTL** [comment]
Very good to hear that Redox would maintain it and doing so within the Redox OS organization on Github makes sense.

I will ask Thomas to join this discussion and take steps forward.

Personally, I want to get this to the point where one can confidently replace OpenSSH with ```redox-ssh``` at some point.

**Comment by MggMuggins** [comment]
Yes, I'd like to see that also. Ssh is one of the most basic tools a server/embedded system needs to be able to handle, I'd say it's got fairly high priority. I don't have a ton of experience using it beyond basic stuff (logging in, sftp, etc), so it'll take me some learning to really dive into the internals and be helpful, but I'm willing to take the time!

**Comment by MggMuggins** [comment]
@ERnsTL Do you have a chat.redox.org account yet? That's a super helpful what to communicate. I'll create an 'ssh' channel now so that we're ready when you show up!

**Comment by ERnsTL** [comment]
I don't have deep knowledge of SSH internals either, only usage experience in various scenarios and use cases, PF, reverse PF, subsystems, key handling, connection keepalive, locking things down, parsing ```authorized_keys``` files etc. etc. that has accumulated over the last 17 years.

This will surely be a nice learning experience as well.

Account on Redox dev chat is set up, I joined the ssh channel there.

**Comment by ERnsTL** [comment]
Just noticed that the project has been transferred to the redox-os organization. Nice!

@jackpot51 I guess this repo will also be moved into the Redox Gitlab? Thus, continue development there..?

**Comment by jackpot51** [comment]
Yep
